src/app/pages/waiting-room-to-car/components/vehicle-registration/vehicle-registration.ts

```diff
@@ -27,14 +39,77 @@ export class VehicleRegistrationComponent implements OnChanges {
   @Output()
   vehicleRegistrationBlur = new EventEmitter<string>();
 
+  @Output()
+  alternateEvidenceChange = new EventEmitter<boolean>();
+  @Output()
+  alternativeEvidenceDescriptionUpdate = new EventEmitter<string>();
+
   formControl: UntypedFormControl;
 
+  motData: MotDataWithStatus = null;
+
+  modalData: string = null;
+
+  hasCalledMOT: boolean = false;
+
+  showSearchSpinner: boolean = false;
+
+  modalRepeatCount: number = 2;
+
   readonly registrationNumberValidator: FieldValidators = getRegistrationNumberValidator();
 
+  constructor(
+    public motApiService: VehicleDetailsApiService,
+    public modalController: ModalController,
+    protected networkState: NetworkStateProvider,
+  ) {
+  }
+
   get invalid(): boolean {
     return !this.formControl.valid && this.formControl.dirty;
   }
 
+  clearData() {
+    this.formGroup.removeControl('evidenceDescriptionCtrl');
+    this.formGroup.removeControl('alternateEvidenceCtrl');
+    this.alternateEvidenceChange.emit(undefined);
+    this.alternativeEvidenceDescriptionUpdate.emit(undefined);
+  }
+
+  getMOT(value: string) {
+    this.clearData();
+    this.hasCalledMOT = false;
+    this.showSearchSpinner = true;
+    this.motApiService.getVehicleByIdentifier(value).subscribe(async (val) => {
+      this.motData = val;
+      // If the MOT is invalid, open the reconfirm modal
+      if (this.motData?.data?.status === MotStatusCodes.NOT_VALID && this.modalRepeatCount !== 0) {
+        this.modalRepeatCount -= 1;
+        await this.loadModal();
+        if (this.modalData !== this.motData.data.registration) {
+          // Call the MOT service again if the new registration is different.
+          this.vehicleRegistration = this.modalData;
+          this.getMOT(this.modalData);
+          return;
+        }
+      }
+      this.hasCalledMOT = true;
+      this.showSearchSpinner = false;
+      this.modalRepeatCount = 2;
+    });
+  }
```

Comment:
This does not follow the ngrx code flow that the rest of the app does i.e.

An action is dispatched
The effect listens out for the action and runs some functionality - in this instance hitting the EP
The reducer then updates state
Page then pipes values from store
Passes those values into `dumb` components

Reply:
This also does not capture the errors the HTTP call could produce so would have no visibility of errors via app logs